run.py

```diff
@@ -50,56 +51,67 @@ def run_one_step_with_cudastreams(func, streamcount):
-      end_event = torch.cuda.Event(enable_timing=True)
+        torch.cuda.synchronize()
+        start_event = torch.cuda.Event(enable_timing=True)
+        end_event = torch.cuda.Event(enable_timing=True)
 
-      start_event.record()
+        start_event.record()
 
-      t0 = time.time()
-      func()
-      t1 = time.time()
+        t0 = time.time()
+        func()
+        t1 = time.time()
 
-      end_event.record()
-      torch.cuda.synchronize()
-      print(f"Ran in {t1 - t0} seconds, gpu time {start_event.elapsed_time(end_event)}.")
+        end_event.record()
+        torch.cuda.synchronize()
+        print(f"Ran in {t1 - t0} seconds, gpu time {start_event.elapsed_time(end_event)}.")
 
     else:
 
-      t0 = time.time()
-      func()
-      t1 = time.time()
+        t0 = time.time()
+        func()
+        t1 = time.time()
 
-      print(f"Ran in {t1 - t0} seconds.")
+        print(f"Ran in {t1 - t0} seconds.")
 
 
 def profile_one_step(func, nwarmup=3):
-    for i in range(nwarmup):
-        func()
-
-    use_cuda = args.device == "cuda"
-
-    with profiler.profile(record_shapes=True, use_cuda = use_cuda) as prof:
-        func()
+    activity_groups = []
+    if args.device == "cuda":
+        activity_groups.append(profiler.ProfilerActivity.CUDA)
+    activity_groups.append(profiler.ProfilerActivity.CPU)
+
+    with profiler.profile(
+        schedule=profiler.schedule(wait=0, warmup=nwarmup, active=1),
+        activities=activity_groups,
+        record_shapes=True,
+        on_trace_ready=profiler.tensorboard_trace_handler(args.profile_folder)
+    ) as prof:
+        for _i in range(nwarmup + 1):
+            func()
+            prof.step()
 
     print(prof.key_averages(group_by_input_shape=True).table(sort_by="cpu_time_total", row_limit=30))
+    print(f"Saved TensorBoard Profiler traces to {args.profile_folder}")
+
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(__doc__)
     parser.add_argument("model", help="Full or partial name of a model to run.  If partial, picks the first match.")
-    parser.add_argument("-d", "--device", choices=["cpu",  "cuda"], default="cpu", help="Which device to use.")
-    parser.add_argument("-m", "--mode", choices=["eager",  "jit"], default="eager", help="Which mode to run.")
-    parser.add_argument("-t", "--test", choices=["eval",  "train"], default="eval", help="Which test to run.")
+    parser.add_argument("-d", "--device", choices=["cpu", "cuda"], default="cpu", help="Which device to use.")
+    parser.add_argument("-m", "--mode", choices=["eager", "jit"], default="eager", help="Which mode to run.")
+    parser.add_argument("-t", "--test", choices=["eval", "train"], default="eval", help="Which test to run.")
     parser.add_argument("--profile", action="store_true", help="Run the profiler around the function")
+    parser.add_argument("--profile-folder", default="./logs", help="Save profiling model traces to this directory.")
```

Comment:
Please add "logs/" to `.gitignore` to prevent future possible commits of the directory.

Reply:
Will do!